gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1318,7 +1318,7 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
             task.setCodegenClasspath(project.getConfigurations().getByName(PEGASUS_PLUGIN_CONFIGURATION));
             task.setSnapshotCompatLevel(PropertyUtil.findCompatLevel(project, FileCompatibilityType.SNAPSHOT));
 
-            task.onlyIf(t -> isPropertyTrue(project, SKIP_IDL_CHECK));
```

Comment:
This is on purpose. 

The plugin has two ways to do compatibility checking:
1. Use current IDL + PDL against prev.snapshot. This is done by two tasks: checkIdl + checkRestModel
2. Use only snapshot files. This is done by checkSnapshot

Normally the plugin uses only option 1. If the skip idl flag is set, it will disable the tasks in option 1 and run the checkSnapshot task instead.

See Lines 1302, 1339, 1358-1366, 1381-1390 for relevant code.

Reply:
I see, thanks for the details.